core/src/main/java/com/microsoft/applicationinsights/internal/config/JaxbAppInsightsConfigurationBuilder.java

```diff
@@ -44,7 +44,11 @@ public ApplicationInsightsXmlConfiguration build(String filename) {
 
             return applicationInsights;
         } catch (JAXBException e) {
-            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR, "Failed to parse configuration file: '%s'", e.getMessage());
+                if (e.getCause() != null) {
+                InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR, "Failed to parse configuration file '%s': '%s'", filename, e.getCause().getMessage());
```

Comment:
1. Indentation
2. Please make sure that this would print the right message in case the XML contains forbidden leading characters or just unexpected tags